### ISSUE TITLE
TypeError in groupby(by=lambda x:x[1]<3) with MultiIndex

### ISSUE THREAD

**Issue by agardelein** [modin-project/modin#756]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Debian 9 unstable
- **Modin installed from (source or binary)**: pip3 install -U git+https://github.com/modin-project/modin
- **Modin version**: 0.5.4
- **Pandas version**: 0.24.2
- **IPython version**: 5.8.0
- **Python version**: 3.7.3
- **Exact command to reproduce**:

### Describe the problem

When trying to perform a groupby based on value comparison of second element of index, TypeError is raised for modin.pandas.

File test.dat:
```
p0,p1,data0,data1,data2
A,0,10,11,12
B,1,20,21,22
C,2,30,31,32
D,3,40,41,42
E,4,50,51,52

```
### Source code / logs
Source code:
```
import modin.pandas
import pandas as pd

# Working example with Pandas
csv1=pd.read_csv('test.dat',header=0,index_col=[0,1])
g1=csv1.groupby(by=lambda x:x[1]<3)
g1.get_group(True)

# Testing with modin.pandas
csv2=modin.pandas.read_csv('test.dat',header=0,index_col=[0,1])
g2=csv2.groupby(by=lambda x:x[1]<3)
```

Error message:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-16-e142f2bb04bf> in <module>()
----> 1 g2=csv2.groupby(by=lambda x:x[1]<3)

~/.local/lib/python3.7/site-packages/modin/pandas/dataframe.py in groupby(self, by, axis, level, as_index, sort, group_keys, squeeze, observed, **kwargs)
    350         idx_name = None
    351         if callable(by):
--> 352             by = by(self.index)
    353         elif isinstance(by, string_types):
    354             idx_name = by

<ipython-input-16-e142f2bb04bf> in <lambda>(x)
----> 1 g2=csv2.groupby(by=lambda x:x[1]<3)

TypeError: '<' not supported between instances of 'tuple' and 'int'
```

**Comment by devin-petersohn** [comment]
Hi @agardelein thanks for the report!

I can reproduce the issue, and it's a logic error on our part. Pandas documentation states that the function is applied to every element, but we have implemented it as being called on the entire index. The line in question should be: `by = self.index.map(by)` instead of `by = by(self.index)`.

Thanks again for the report! We are in the middle of a large internal rewrite for performance, so once that is done we can get to this.

**Comment by agardelein** [comment]
Fixing it on my tree lead to following when performing a `groupby` followed by a `get_group`.

Note that the `print` inserted at line 1836 use `self.columns` since `self.keys()` at line 395 is supposed to return `self.columns`.

```
In [6]: df_g=csv2.groupby('p0').get_group('A')
p0 Index(['data0', 'data1', 'data2'], dtype='object')
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-6-704c3a79c97b> in <module>()
----> 1 df_g=csv2.groupby('p0').get_group('A')

~/.local/lib/python3.7/site-packages/modin/pandas/dataframe.py in groupby(self, by, axis, level, as_index, sort, group_keys, squeeze, observed, **kwargs)
    353         elif isinstance(by, string_types):
    354             idx_name = by
--> 355             by = self.__getitem__(by)._query_compiler
    356         elif is_list_like(by):
    357             if isinstance(by, Series):

~/.local/lib/python3.7/site-packages/modin/pandas/base.py in __getitem__(self, key)
   3220             return self._getitem_slice(indexer)
   3221         else:
-> 3222             return self._getitem(key)
   3223 
   3224     def _getitem_slice(self, key):

~/.local/lib/python3.7/site-packages/modin/pandas/dataframe.py in _getitem(self, key)
   1835         else:
   1836             print(key, self.columns)
-> 1837             return self._getitem_column(key)
   1838 
   1839     def _getitem_column(self, key):

~/.local/lib/python3.7/site-packages/modin/pandas/dataframe.py in _getitem_column(self, key)
   1839     def _getitem_column(self, key):
   1840         if key not in self.keys():
-> 1841             raise KeyError("{}".format(key))
   1842         s = self._reduce_dimension(self._query_compiler.getitem_column_array([key]))
   1843         s._parent = self

KeyError: 'p0'
```

**Comment by devin-petersohn** [comment]
@agardelein, thanks for the update, the MultiLevelIndex support in Modin needs some work.

When you use a label for the `groupby`, pandas will also allow you to specify a label from the index if is it a MultiLevelIndex. This is not documented, which is why it was missed.

It's worth tracking as a separate issue since it's not directly related to the bug in this issue. I can create that issue if you'd like.

**Comment by devin-petersohn** [comment]
Closing as this should be solved by #757. Please re-open if the issue is still there.
